docker-environment-driver/src/main/java/org/jboss/pnc/environment/docker/DockerEnvironmentDriver.java

```diff
@@ -1,17 +1,297 @@
 package org.jboss.pnc.environment.docker;
 
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.MediaType;
+
+import org.jboss.pnc.model.BuildType;
 import org.jboss.pnc.model.Environment;
+import org.jboss.pnc.model.OperationalSystem;
 import org.jboss.pnc.spi.environment.EnvironmentDriver;
+import org.jboss.pnc.spi.environment.RunningEnvironment;
+import org.jboss.pnc.spi.environment.exception.EnvironmentDriverException;
+import org.jboss.resteasy.client.ClientRequest;
+import org.jboss.resteasy.client.ClientResponse;
+import org.jclouds.ContextBuilder;
+import org.jclouds.compute.ComputeServiceContext;
+import org.jclouds.docker.DockerApi;
+import org.jclouds.docker.domain.Config;
+import org.jclouds.docker.domain.Container;
+import org.jclouds.docker.domain.HostConfig;
+import org.jclouds.docker.features.RemoteApi;
+import org.jclouds.domain.LoginCredentials;
+import org.jclouds.http.handlers.BackoffLimitedRetryHandler;
+import org.jclouds.io.Payloads;
+import org.jclouds.logging.slf4j.config.SLF4JLoggingModule;
+import org.jclouds.ssh.SshClient;
+import org.jclouds.sshj.SshjSshClient;
+import org.jclouds.sshj.config.SshjSshClientModule;
 
-import java.lang.invoke.MethodHandles;
-import java.util.logging.Logger;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.net.HostAndPort;
+import com.google.inject.Module;
+import com.jcraft.jsch.agentproxy.Connector;
 
+/**
+ * Implementation of environment driver, which uses Docker to run environments
+ * 
+ * @author Jakub Bartecek <jbartece@redhat.com>
+ *
+ */
+@ApplicationScoped
 public class DockerEnvironmentDriver implements EnvironmentDriver {
 
-    private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass().getName());
+    private static final Logger logger =
+            Logger.getLogger(MethodHandles.lookup().lookupClass().getName());
+
+    /** User in running environment to which we can connect with SSH */
```

Comment:
Yeah, this is just my preference... Instead of using comments I would prefer more expressive variable names (even though they are much longer in that case). Comments also tend to survive refactorings and after some time they don't reflect the reality.